File: packages/sdk/src/super-admin-command-center.ts
/** Typed SDK for Domain 74 — Super Admin Command Center, Feature Flags, Audit, Platform Overrides. */

export type SaRole          = 'viewer' | 'sa_operator' | 'sa_admin' | 'sa_root';
export type FlagStatus      = 'draft' | 'active' | 'paused' | 'archived';
export type OverrideScope   = 'platform' | 'tenant' | 'user' | 'feature' | 'route' | 'domain';
export type OverrideKind    = 'rate_limit' | 'maintenance' | 'config' | 'entitlement' | 'kill_switch' | 'dark_launch' | 'cost_cap' | 'rollout';
export type OverrideStatus  = 'active' | 'paused' | 'expired' | 'archived';
export type IncidentSeverity= 'sev1' | 'sev2' | 'sev3' | 'sev4';
export type IncidentStatus  = 'open' | 'mitigated' | 'resolved' | 'postmortem' | 'archived';

export interface FlagSegment { kind: string; value: string }
export interface FlagVariant { key: string; weight: number; payload: Record<string, unknown> }

export interface FeatureFlag {
  id: string; key: string; name: string; description: string;
  enabled: boolean; rollout_pct: number; status: FlagStatus;
  environments: string[]; segments: FlagSegment[]; variants: FlagVariant[];
  owner_id?: string | null; created_by?: string | null;
  created_at: string; updated_at: string;
}
export interface PlatformOverride {
  id: string; scope: OverrideScope; scope_id?: string | null;
  kind: OverrideKind; value: Record<string, unknown>;
  reason: string; status: OverrideStatus;
  created_by?: string | null; expires_at?: string | null;
  created_at: string; updated_at: string;
}
export interface PlatformIncident {
  id: string; title: string; severity: IncidentSeverity; status: IncidentStatus;
  scope: string; commander?: string | null;
  opened_at: string; mitigated_at?: string | null; resolved_at?: string | null;
  notes: string; meta: Record<string, unknown>;
}
export interface SaAuditEvent {
  id: string; actor_id?: string | null; domain: string;
  target_id?: string | null; action: string;
  from_state?: string | null; to_state?: string | null;
  diff: Record<string, unknown>; ip?: string | null; user_agent?: string | null;
  created_at: string;
}
export interface SaKpis {
  flagsByStatus:     Partial<Record<FlagStatus, number>>;
  overridesByStatus: Partial<Record<OverrideStatus, number>>;
  overridesByKind:   Partial<Record<OverrideKind, number>>;
  incidentsByStatus: Partial<Record<IncidentStatus, number>>;
  openIncidentsBySev:Partial<Record<IncidentSeverity, number>>;
  auditEvents24h:    number;
  killSwitchesActive:number;
}
export interface SaInsight { id: string; severity: 'success' | 'info' | 'warn' | 'critical'; title: string }
export interface SaOverview {
  kpis: SaKpis;
  flagsActive: FeatureFlag[];
  overridesActive: PlatformOverride[];
  openIncidents: PlatformIncident[];
  recentAudit: SaAuditEvent[];
  insights: SaInsight[];
  computedAt: string;
}
export interface SaList<T> { items: T[]; total: number; meta: { source: string; role: SaRole; page: number; pageSize: number } }

File: apps/analytics-python/app/super_admin_command_center.py
"""Domain 74 — Super Admin Command Center insights (deterministic)."""
from __future__ import annotations
from datetime import datetime, timezone
from typing import Any, Dict, List

from fastapi import APIRouter
from pydantic import BaseModel, Field

router = APIRouter(prefix="/super-admin-command-center", tags=["super-admin-command-center"])


class InsightsIn(BaseModel):
    signals: Dict[str, Any] = Field(default_factory=dict)


def _build(s: Dict[str, Any]) -> List[Dict[str, Any]]:
    out: List[Dict[str, Any]] = []
    open_sev = s.get("openIncidentsBySev") or {}
    sev1 = int(open_sev.get("sev1") or 0)
    sev2 = int(open_sev.get("sev2") or 0)
    kills = int(s.get("killSwitchesActive") or 0)
    audit = int(s.get("auditEvents24h") or 0)
    draft = int((s.get("flagsByStatus") or {}).get("draft") or 0)
    expired_overrides = int((s.get("overridesByStatus") or {}).get("expired") or 0)
    if sev1:    out.append({"id": "sev1_open",      "severity": "critical",
                            "title": f"{sev1} sev1 incident(s) open — page on-call."})
    if sev2:    out.append({"id": "sev2_open",      "severity": "warn",
                            "title": f"{sev2} sev2 incident(s) open."})
    if kills:   out.append({"id": "kill_switches",  "severity": "critical",
                            "title": f"{kills} kill-switch override(s) active."})
    if audit > 200:
        out.append({"id": "audit_volume", "severity": "info",
                    "title": f"{audit} admin events in 24h — review for anomalies."})
    if draft > 5:
        out.append({"id": "draft_flags",  "severity": "info",
                    "title": f"{draft} feature flag(s) in draft."})
    if expired_overrides:
        out.append({"id": "expired_overrides", "severity": "warn",
                    "title": f"{expired_overrides} override(s) expired — clean up."})
    if not out:
        out.append({"id": "platform_healthy", "severity": "success",
                    "title": "Platform posture healthy."})
    return out


@router.post("/insights")
def insights(body: InsightsIn):
    return {"insights": _build(body.signals),
            "meta": {"computed_at": datetime.now(timezone.utc).isoformat()}}
